File: src/imu_sub.cpp
#include <ros/ros.h>
#include <sensor_msgs/Imu.h>

int count = 0;
ros::Publisher pub;
ros::Time last_pub;

void callback(const sensor_msgs::Imu::ConstPtr& msg)
{
  ros::Time cur = ros::Time::now();
  ros::Duration period(1.0);
  ros::Duration elapsed = cur - last_pub;
  ros::Duration remaining = period - elapsed;
  if (remaining > 0)
  {
    remaining.sleep();
    return;
  }
  pub.publish(msg);
  last_pub = cur;

  count += 1;
  if (count % 50 == 0)
  {
    //ROS_INFO_STREAM(msg);
  }
}

int main(int argc, char** argv)
{
  ros::init(argc, argv, "imu_sub");
  ros::NodeHandle nh;
  last_pub = ros::Time::now();
  pub = nh.advertise<sensor_msgs::Imu>("imu_throttle", 2);
  ros::Subscriber sub = nh.subscribe("imu", 1, callback);
  ros::spin();
}

File: scripts/imu_sub.py
#!/usr/bin/env python

import rospy

from sensor_msgs.msg import Imu

def callback(data):
    global last_pub  # rate
    global start
    global period
    cur = rospy.Time.now()
    # elapsed = data.header.stamp - last_pub
    elapsed = cur - last_pub
    remaining = period - elapsed
    if remaining.to_sec() > 0:
        rospy.sleep(remaining.to_sec())
        return

    global count
    count += 1
    global pub
    pub.publish(data)
    # last_pub = data.header.stamp
    last_pub = cur
       # rate.sleep()

rospy.init_node('imu_sub')

count = 0
last_pub = rospy.Time.now()
start = rospy.Time.now()
period = 0.0
rate = rospy.get_param("~rate", 0.0)
if (rate <= 0.0):
    rate = 1.0
period = rospy.Duration(1.0 / rate)

sub = rospy.Subscriber("imu", Imu, callback, queue_size=1)
pub = rospy.Publisher("imu_throttle", Imu, queue_size=2)
rospy.spin()
